book/bayesian_optimization.qmd

```diff
@@ -0,0 +1,844 @@
+
+In the worst-case (all iterations erroring out), the BO algorithm will therefore simply perform a random search.
+Ideally, the learner wrapped within the surrogate makes use of encapsulation and can rely on a fallback learner (see @sec-encapsulation-fallback) that will jump into action before this final safety net of proposing the next candidate uniformly at random is triggered.
+Note that the value of the acquisition function is always also logged into the archive of the optimization instance.
+To make sure that the BO algorithm behaved as expected, users should always inspect the log of the optimization process and inspect the archive and check whether the acquisition function column is populated as expected.
+
+`r mlr3mbo` tries to use 'intelligent' defaults regarding the choice of surrogate model, acquisition function, acquisition function optimizer and even the loop function.
+To see an up-to-date overview of these defaults, users should inspect the following man page:
+
+```{r}
+?mbo_defaults
+```
+
+For example, in the case of a purely numeric search space, `r mlr3mbo` will by default use a Gaussian Process as surrogate model and a random forest as fallback learner and additionally encapsulates the learner via the `r ref_pkg("evaluate")` package.
+As a result of defaults existing for all building blocks, users can perform BO without specifying any building blocks and can still expect decent optimization performance.
+
+In practice users may prefer a more robust BO variant over a potentially better performing but unstable variant.
+For example, Gaussian Processes are highly subject to the choice of kernel but also the kernel parameters themselves which are usually obtained via maximum likelihood estimation.
+Suboptimal parameter values can result in white noise models with a constant mean and variance prediction (except for the interpolation of training data).
+In this case, the surrogate model will not provide useful mean and variance predictions resulting in poor performance of the BO algorithm.
+Moreover, fitting a vanilla Gaussian Process scales cubic in the number of data points and therefore the overhead of the BO algorithm grows with the number of iterations.
+Besides, vanilla Gaussian Processes natively cannot handle categorical input variables or dependencies in the search space (recall that in HPO we often deal with hierarchical mixed spaces).
+In contrast, a random forest is cheap to train, quite robust in the sense that it is not that sensitive to its hyperparameters as a Gaussian Process and can easily handle hierarchical mixed spaces.
+On the downside, a random forest is not really Bayesian (i.e., there is no posterior predictive distribution) and suffers from poor uncertainty estimates and poor extrapolation.
+Nevertheless, random forests usually perform quite well as a surrogate model in BO.
```

Comment:
Unsure how much of this detail is really needed and useful here, it is very technical. On the other hand some readers might really like that